static/js/components/forms/ProfileFormFields.js

```diff
@@ -0,0 +1,445 @@
+                ) : null}
+            </div>
+          )}
+        />
+      </div>
+      <div className="form-group">
+        <label className="font-weight-bold">Country*</label>
+        <Field
+          component="select"
+          name="legal_address.country"
+          className="form-control"
+          onChange={e => {
+            setFieldValue("legal_address.country", e.target.value)
+            setFieldTouched("legal_address.country")
+            if (!includes(e.target.value, [US_ALPHA_2, CA_ALPHA_2])) {
+              setFieldValue("legal_address.state_or_territory", "")
+              setFieldValue("legal_address.postal_code", "")
+            }
+          }}
+        >
+          <option value="">-----</option>
+          {countries
+            ? countries.map((country, i) => (
+              <option key={i} value={country.code}>
+                {country.name}
+              </option>
+            ))
+            : null}
+        </Field>
+        <ErrorMessage name="legal_address.country" component={FormError} />
+      </div>
+      {includes(values.legal_address.country, COUNTRIES_REQUIRING_STATE) ? (
+        <div className="form-group">
+          <label className="font-weight-bold">State/Province*</label>
+          <Field
+            component="select"
+            name="legal_address.state_or_territory"
+            className="form-control"
+          >
+            <option value="">-----</option>
+            {find(
+              propEq("code", values.legal_address.country),
+              countries
+            ).states.map((state, i) => (
+              <option key={i} value={state.code}>
+                {state.name}
+              </option>
+            ))}
+          </Field>
+          <ErrorMessage
+            name="legal_address.state_or_territory"
+            component={FormError}
+          />
+        </div>
+      ) : null}
+      <div className="form-group">
+        <label htmlFor="legal_address.city" className="font-weight-bold">
+          City*
+        </label>
+        <Field type="text" name="legal_address.city" className="form-control" />
+        <ErrorMessage name="legal_address.city" component={FormError} />
+      </div>
+      {includes(
+        values.legal_address.country,
+        COUNTRIES_REQUIRING_POSTAL_CODE
+      ) ? (
+          <div className="form-group">
+            <label
+              htmlFor="legal_address.postal_code"
+              className="font-weight-bold"
+            >
+            Zip/Postal Code*
+            </label>
+            <Field
+              type="text"
+              name="legal_address.postal_code"
+              className="form-control"
+            />
+            <ErrorMessage
+              name="legal_address.postal_code"
+              component={FormError}
+            />
+          </div>
+        ) : null}
+    </React.Fragment>
+  )
+}
+
+export function renderProfileFields() {
+  const seedYear = moment().year()
```

Comment:
This should probably be a module level constant, no need to recompute this every time we re-render.